src/geometry/euclidean.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2020 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Joseph Myers.
+-/
+import algebra.big_operators
+import analysis.normed_space.real_inner_product
+import analysis.normed_space.add_torsor
+
+noncomputable theory
+open_locale big_operators
+
+/-!
+# Euclidean spaces
+
+This file defines Euclidean affine spaces.
+
+## Implementation notes
+
+Rather than requiring Euclidean affine spaces to be finite-dimensional
+(as in the definition on Wikipedia), this is specified only for those
+theorems that need it.
+
+## References
+
+* https://en.wikipedia.org/wiki/Euclidean_space
+
+-/
+
+section prio
+set_option default_priority 100 -- see Note [default priority]
+/-- A `euclidean_affine_space V P` is an affine space with points `P`
+over an `inner_product_space V`. -/
+class euclidean_affine_space (V : Type*) (P : Type*) [inner_product_space V] [metric_space P]
+  extends normed_add_torsor V P
+end prio
+
+/-- The standard Euclidean space, fin n → ℝ. -/
+instance standard_euclidean_space_is_vector_space (n : ℕ) : vector_space ℝ (fin n → ℝ) :=
+by apply_instance
```

Comment:
Either delete, or document that it is helpful for short-circuiting typeclass search.

(Personally, I'd allow an `example` here to verify that the typeclass search is working as expected, but that's not usual mathlib practice.)

Reply:
Example is fine, but please don't keep this instance.

Reply:
With the change to using `euclidean_space n` as a copy of the type, I believe this instance is now necessary (and it also has a comment on it).